### PR TITLE
Add Code Repository Flag & Support PR Decoration For GH/GL On-prem (AST-72979, AST-72975)

### PR DESCRIPTION
**Pull request by miryamfoiferCX**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> 1. add code-repository-url optional flag
> 2. support PR decoration command for github & gitlab on-prem

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> unit test for the new optional flag
>
> manually E2E tests on on-prem machines

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: .github/workflows/pr-linter.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
             exit 1
           fi
           
-          if ! [[ "$PR_TITLE" =~ \(AST-[0-9]+\)$ ]]; then
+          if ! [[ "$PR_TITLE" =~ \(AST-[0-9]+\)$ || "$PR_TITLE" =~ \(AST-[0-9]+(, AST-[0-9]+)*\)$ ]]; then
```
Comment (OrShamirCM) [inline .github/workflows/pr-linter.yml:21]: nice!

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM cgr.dev/chainguard/bash@sha256:f8e48690d991e6814c81f063833176439e8f0d4bc1c5f0a47f94858dea3e4f44
+FROM cgr.dev/chainguard/bash@sha256:e1d16dec8d976859080d984167109b3557c2b6494f10be08147806b78bdef691
```
Comment (OrShamirCM) [inline Dockerfile:1]: I think that Andre upgrade the versio, no?

File: internal/params/flags.go [hunk 3]
```diff
@@ -33,6 +33,8 @@ const (
 	BranchFlag                   = "branch"
 	BranchFlagSh                 = "b"
 	ScanIDFlag                   = "scan-id"
+	CodeRepositoryFlag           = "code-repository-url"
```
Comment (OrShamirCM) [inline internal/params/flags.go:36]: didn't it changed to repository-sever-url??

File: internal/params/flags.go [hunk 4]
```diff
@@ -33,6 +33,8 @@ const (
...
+	CodeRepositoryFlagUsage      = "Code repository URL (optional for self-hosted Azure DevOps)"
```
Comment (OrShamirCM) [inline internal/params/flags.go:37]: it's only for Azure? not for any pr deco?

File: test/integration/pr_test.go [hunk 5]
```diff
@@ -51,6 +56,46 @@ func TestPRGithubDecorationSuccessCase(t *testing.T) {
...
+		flag(params.RepoNameFlag),
+		os.Getenv(prGithubRepoName),
+		flag(params.CodeRepositoryFlag),
+		"https://github.example.com",
+	}
+
+	monkey.Patch((*wrappers.PRHTTPWrapper).PostPRDecoration, func(*wrappers.PRHTTPWrapper, *wrappers.PRModel) (string, *wrappers.WebError, error) {
+		return githubPRCommentCreated, nil, nil
+	})
+	defer monkey.Unpatch((*wrappers.PRHTTPWrapper).PostPRDecoration)
```
Comment (OrShamirCM) [inline test/integration/pr_test.go:82]: nice!

File: test/integration/pr_test.go [hunk 6]
```diff
@@ -51,6 +56,46 @@ func TestPRGithubDecorationSuccessCase(t *testing.T) {
 	assert.NilError(t, err, "Error should be nil")
 }
 
+func TestPRGithubOnPremDecorationSuccessCase(t *testing.T) {
```
Comment (OrShamirCM) [inline test/integration/pr_test.go:59]: what are you checking here?
try to uw please when_given_then(should)

File: test/integration/pr_test.go [hunk 7]
```diff
@@ -95,6 +140,50 @@ func TestPRGitlabDecorationSuccessCase(t *testing.T) {
 	assert.NilError(t, err, "Error should be nil")
 }
 
+func TestPRGitlabOnPremDecorationSuccessCase(t *testing.T) {
```
Comment (OrShamirCM) [inline test/integration/pr_test.go:143]: same here  ( test title)

File: test/integration/pr_test.go (hunk 6 above)
Comment (miryamfoiferCX) [reply test/integration/pr_test.go:79]: TestPRGithubDecoration_WhenUseCodeRepositoryFlag_ShouldSuccess

File: internal/params/flags.go (hunk 3 above)
Comment (miryamfoiferCX) [reply internal/params/flags.go:36]: waiting for product to answer....

File: Dockerfile (hunk 2 above)
Comment (miryamfoiferCX) [reply Dockerfile:1]: no, he didn't touch the dockerfile
see his PR:
https://github.com/Checkmarx/ast-cli/pull/923/files

File: internal/params/flags.go (hunk 4 above)
Comment (miryamfoiferCX) [reply internal/params/flags.go:37]: changed to 
Code repository URL (optional for self-hosted SCMs)

**Review by OrShamirCM — APPROVED** [review]

**Review by OrShamirCM — DISMISSED** [review]

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**